### PR TITLE
Support Parameter Set For Proper Tab Completion

### PR DESCRIPTION
**Pull request by JacobErnst98**
Tab Completion works best with parameter sets.

### CODE REVIEW

File: endpoints/Agents.ps1 [hunk 1]
```diff
@@ -1,28 +1,33 @@
...
+    # Machine Name; Default hostname of PC
+    [Parameter(Mandatory=$false,ParameterSetName=’MachineName’)][string]$MachineName=$env:COMPUTERNAME
   )
-  return New-GetRequest -Endpoint "/agents/$AgentID"
+  if($ID){
+    return New-GetRequest -Endpoint "/agents/$AgentID"
+  }
+  if($MachineName){
+    return New-GetRequest -Endpoint "/agents/machine/$MachineName"
+  }
```
Comment (davejlong) [inline endpoints/Agents.ps1]: You're using an `$AgentID` variable but not setting one... which upon further review is also what I had originally. So the `$AgentID` variable needs to be changed to `$ID`.

**Review by davejlong — CHANGES_REQUESTED** [review]
I put a comment in the code, but also my preference is that `if` statements be formatted as:

```
if (...) {
  ...
} else {
  ...
}
```

File: endpoints/Agents.ps1 (hunk 1 above)
Comment (JacobErnst98) [reply endpoints/Agents.ps1]: Fixed in latest Push

**Comment by JacobErnst98** [comment]
> I put a comment in the code, but also my preference is that `if` statements be formatted as:
> 
> ```
> if (...) {
>   ...
> } else {
>   ...
> }
> ```

If we're referring to 
```
if($AgentID){
    return New-GetRequest -Endpoint "/agents/$AgentID"
  }
  if($MachineName){
    return New-GetRequest -Endpoint "/agents/machine/$MachineName"
  }
```
There is no reason to use an else as the return are in place, if we are referring to other code my apologies.

**Review by JacobErnst98 — COMMENTED** [review]
Fixed in Latest Commit

**Review by JacobErnst98 — COMMENTED** [review]
Fixed in latest commit

**Review by davejlong — APPROVED** [review]
All of your single quotes are causing an error. They're not standard apostrophe characters, but quotation marks. Also, I'd prefer to use

**Comment by davejlong** [comment]
Thanks for the PR! I'll include it in the next release which should be out this week along with an endpoint for the Custom Values API endpoint.

**Pull request merged**